### PR TITLE
Extract clean workspace connection provider

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/core/_connection_provider/_workspace_connection_provider.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+        try:
+            from azure.core.exceptions import ClientAuthenticationError
+
+            from ._models import WorkspaceConnectionPropertiesV2BasicResource
+        except ImportError as e:
+            raise MissingRequiredPackage(
+                message="Please install 'azure-identity>=1.12.0,<2.0.0' and 'msrest' to use workspace connection."
+            ) from e
+        try:
+            from promptflow.azure._utils.general import get_arm_token
```
Comment (brynn-code) [inline src/promptflow/promptflow/core/_connection_provider/_workspace_connection_provider.py:238]: Mark: Remove this

Comment (wangchao1230) [inline src/promptflow/promptflow/core/_connection_provider/_workspace_connection_provider.py:238]: shall we do a try import first?

File: src/promptflow/promptflow/core/_connection_provider/_workspace_connection_provider.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+            name,
+            subscription_id=self.subscription_id,
+            resource_group_name=self.resource_group_name,
+            workspace_name=self.workspace_name,
+            credential=self.credential,
+        )
+        return _Connection._from_execution_connection_dict(name=name, data=connection_dict)
+
+
+class AccessDeniedError(UserErrorException):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/core/_connection_provider/_workspace_connection_provider.py:294]: shall we extract a seprate errors file

File: src/promptflow/promptflow/core/_connection_provider/_workspace_connection_provider.py (hunk 1 above)
Comment (brynn-code) [reply src/promptflow/promptflow/core/_connection_provider/_workspace_connection_provider.py]: Updated for the token related part.

Comment (brynn-code) [reply src/promptflow/promptflow/core/_connection_provider/_workspace_connection_provider.py]: use try import azure-ai-ml for now.

**Review by crazygao — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**